js/router/SharedNavigator.js

```diff
@@ -77,6 +104,33 @@ const SharedNavigator = React.createClass ({
     });
   },
 
+  _onNotification(notification) {
+    const data = loadCachedFormDataById(notification.data.formId);
+    const path = {path: 'form', title: data.survey.title, survey: data.survey, form: data.form}
```

Comment:
I would add some defensive coding here in case a Notification comes in from the server for a Form that hasn't been properly cached/recached yet.

Something along the lines of a `return` if the data gathered by `loadCachedFormDataById` is missing a form/survey.

Although that issue I mentioned seems to have been taken care of by a Content Reset, it may happen to users if a Survey gets updated/edited on the admin end and the client has never had a chance to refresh it.